_posts/2020-09-30-intro-to-hive-connector.md

```diff
@@ -0,0 +1,111 @@
+metastore would have taken a long time to repopulate in another format. The
+metastore service still functioned properly and didn't need to be changed, and 
+only the runtime that executed Hive queries needed replacement. So the creators
+took this into account and built the Presto engine to utilize the existing
+metastore metadata and files residing in storage, and the presto runtime
+effectively replaced the Hive runtime responsible for analyzing the data.
+
+![](/assets/blog/intro-to-hive-connector/presto.png)
+
+Notice that the only change in the Presto architecture is the runtime, and
+the Hive metastore still exists along with the storage. This was not by
+accident. This design was to address a common pattern that many companies faced,
+and it simplified the migration from using Hive to using Presto. Since the 
+central model used to store the metadata makes use of the Hive metastore
+regardless of the storage component used, it seemed fitting that the connector 
+received the name Hive connector since the metastore can be configured over
+different storage solutions. 
+
+Where the confusion tends to come from, is when you search for a connector
+from the context of the storage systems you want to query and may not even be 
+aware the metastore is a necessity or even exists. You will typically look for
+something like an S3 connector or GCS connector when all you really need is the
+Hive connector which connects them to a Hive metastore service, configured 
+to save metadata over the filesystem you are writing to. 
+
+To be clear, the Hive metastore service is the only Hive binary used in the
+entire Presto ecosystem when using the Hive connector. The Hive metastore is
+actually a simple service with a binary API using Thrift. This service makes
+updates to the metadata state which gets stored into an RDBMS such as
+PostgreSQL, MySQL, or MariaDB. There are other variations of the Hive metastore
+such as AWS Glue which acts as a drop-in substitution. Nonetheless, Glue and
+other substitutions still rely on the Hive metadata model to store this table
+information.
+
+To drive this point home, I created a Katacoda exercise. In the following
+scenario, I created a docker environment that creates four docker container
+nodes based on the following images: `presto`, `minio`, `hive-metastore`, and 
+`mariadb`. Presto will be _the runtime_ in this scenario that replaces Hive. 
+MinIO is an open-source S3 compatible object storage system that acts as 
+_the storage_ component. For _the metastore_, you will run a `hive-metastore` 
+instance that connects to a `maraidb` instance to store the metadata. You will
+be able to play around with the system and look at the configurations. 
+
+In this Katacoda example, you'll run a docker-compose script that generates all
+of these instances. There's no need to configure these instances or really
+understand how they are set up. The scenario asks you to run a query to populate
```

Comment:
That last sentence is really all you need to say

The Katacoda example teaches to run a SQL query to populate the objects storage MinIO  with some sample data. You can then see the resulting metadata in the MariaDB database backing the Hive metastore.